compiler/src/iree/compiler/Tools/init_iree.cc

```diff
@@ -0,0 +1,39 @@
+#include "iree/compiler/Tools/init_iree.h"
+
+#include "iree/compiler/Tools/version.h"
+#include "llvm/Support/CommandLine.h"
+
+static void versionPrinter(llvm::raw_ostream &os) {
+  os << "IREE (https://iree-org.github.io/):\n  ";
+  if (std::string version = mlir::iree_compiler::getIreeRevision();
+      !version.empty()) {
+    os << "IREE compiler version " << version << "\n  ";
+  } else {
+    os << "IREE compiler version (unknown)\n  ";
+  }
```

Comment:
How bad would it be if version escaped the scope of this "if"? I think it would be clearer on its own line. That also opens up the possibility of a bit less nesting.

```suggestion
  std::string version = mlir::iree_compiler::getIreeRevision()
  if (version.empty()) {
    version = "(unknown)"
  }
  os << "IREE compiler version " << version << "\n  ";
```